### PR TITLE
Jim/76927/remaining-time-component-ts-migration

### PR DESCRIPTION
**Pull request by jim-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ...

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [X] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/components/src/components/remaining-time/remaining-time.tsx [hunk 1]
```diff
@@ -1,9 +1,16 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import { formatDuration, getDiffDuration } from '@deriv/shared';
+import moment from 'moment';
 
-const RemainingTime = ({ end_time = null, format, getCardLabels, start_time }) => {
-    if (!+end_time || start_time.unix() > +end_time) {
+type TRemainingTimeProps = {
+    end_time?: number | null;
```
Comment (yashim-deriv) [inline packages/components/src/components/remaining-time/remaining-time.tsx]: no need to declare type as `null` since the field is already nullable

**Review by yashim-deriv — CHANGES_REQUESTED** [review]

**Review by shayan-deriv — APPROVED** [review]

**Review by yauheni-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]
Except the yashim's comment the rest of the things are LGTM

File: packages/components/src/components/remaining-time/remaining-time.tsx [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+    start_time: moment.Moment;
+    format: string;
+    getCardLabels: () => { [key: string]: string }; // TODO Use the one from common after contract-card migration
```
Comment (hirad-deriv) [inline packages/components/src/components/remaining-time/remaining-time.tsx:9]: I think using Record utility type would be the cleaner choice.
Record<string, string>

Comment (jim-deriv) [reply packages/components/src/components/remaining-time/remaining-time.tsx:9]: Yes @hirad-rewok, but I already typed this `function` is the `contract-card` component, just that there are some few tweeks I need to make in that card and then I push the changes. The `type` is something like `ReturnType<typeof getCardLabels>`, which gives auto complete for the object properties on the returned object

**Review by farzin-deriv — APPROVED** [review]

**Review by maryia-deriv — APPROVED** [review]

File: packages/components/src/components/remaining-time/remaining-time.tsx (hunk 1 above)
Comment (maryia-deriv) [reply packages/components/src/components/remaining-time/remaining-time.tsx]: @jim-deriv  I also can't find this prop to be passed from outside as null. Is there such a place? 🙂

**Review by hirad-deriv — APPROVED** [review]

File: packages/components/src/components/remaining-time/remaining-time.tsx (hunk 1 above)
Comment (jim-deriv) [reply packages/components/src/components/remaining-time/remaining-time.tsx]: Alright guys. I'll remove null from the prop :sweat_smile:

cc: @yashim-deriv, @maryia-binary @farzin-fs

**Review by maryia-deriv — APPROVED** [review]

**Review by likhith-deriv — APPROVED** [review]

**Pull request merged**